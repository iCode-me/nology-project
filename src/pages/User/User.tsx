import { useCallback, useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import Layout from "../../components/Layout/Layout";
import { getUser } from "../../data/api";
import UserData from "../../types/internal/UserData";

// to see this page component action, append this to your URL: "/users/1" 🧐
// TODO: Add section in readme breaking down this component
type UserParams = {
  userId: string;
};

const User: React.FC = () => {
  const { userId } = useParams<UserParams>();
  const [user, setUser] = useState<UserData | null>();

  useEffect(() => {
    if (!userId) return;
    (async () => {
      setUser((await getUser(parseInt(userId))) ?? null);
    })();
  }, [userId]);

  const userTsx = useCallback(
    (user: UserData | null) =>
      user ? (
        <>
          <p>Name : {user?.fullName}</p>
          <p>User Id : {user?.id}</p>
        </>
      ) : (
        <span>Sorry, no user found with id of {userId}</span>
      ),
    [userId]
  );

  return (
    <Layout>
      <h1>User</h1>
      {user !== undefined ? userTsx(user) : <span>Loading...</span>}
    </Layout>
  );
};

export default User;
