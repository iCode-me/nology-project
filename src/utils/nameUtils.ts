export const formatName = (firstName: string, lastName: string): string => {
  return `${firstName} ${lastName}`;
};

// export const formatNameArray = (names: string[]): string => {
//   return names.join(" ");
// };
