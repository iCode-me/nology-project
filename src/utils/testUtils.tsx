import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

type CustomRenderOptions = {
  useRouting: boolean;
};

export const customRender = (
  ui: JSX.Element,
  options: CustomRenderOptions = {
    useRouting: true,
  }
) => {
  let uiResult = ui;

  if (options.useRouting) {
    uiResult = wrapWithRouting(ui);
  }

  return render(uiResult);
};

const wrapWithRouting = (ui: JSX.Element): JSX.Element => {
  return <BrowserRouter>{ui}</BrowserRouter>;
};
