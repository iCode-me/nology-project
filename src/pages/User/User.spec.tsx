import { customRender } from "../../utils/testUtils";
import User from "./User";

export default describe("User Page", () => {
  it("Renders component in loading state", () => {
    const { container } = customRender(<User />);
    expect(container).toMatchSnapshot();
  });

  //TODO: Add mock data to show component with data
});
