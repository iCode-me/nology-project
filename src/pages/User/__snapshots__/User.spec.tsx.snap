// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Page Renders component in loading state 1`] = `
<div>
  <div
    class="layout"
  >
    <main
      class="layout__main"
    >
      <h1>
        User
      </h1>
      <span>
        Loading...
      </span>
    </main>
  </div>
</div>
`;
